### DISCUSSION TITLE
TTS Options?

### DISCUSSION THREAD

**Discussion by bubbabug** [LostRuins/koboldcpp#156] [Ideas]
Are there any configurable options for the TTS function? Currently when I toggle TTS in chat mode it reads back both user input and bot response. Ideally, in this mode I would like for the TTS to only activate when the bot responds. I understand that the current implementation may work better for story writing, but I was wondering if there was a script that could be modified to alter the behavior of the TTS function?

**Comment by LostRuins** [comment]
yeah ideally it should be a bit more customizable, though TTS has been a bit of a lower priority

**Reply by Innomen** [reply]
If you baked some voice cloning into kobold it would be amazing. https://github.com/serp-ai/bark-with-voice-clone Or maybe at least some pretrained speakers?
